light/provider/http/http.go

```diff
@@ -121,7 +130,7 @@ func (p *http) validatorSet(ctx context.Context, height *int64) (*types.Validato
 	)
 
 	for len(vals) != total && page <= maxPages {
-		for attempt := 1; attempt <= p.maxRetryAttempts; attempt++ {
+		for attempt := 0; attempt <= p.maxRetryAttempts; attempt++ {
```

Comment:
```suggestion
		for attempt := 0; attempt < p.maxRetryAttempts; attempt++ {
```

?

Reply:
but in that case if `maxRetryAttempts` is 0 then it won't even make the initial request